### ISSUE TITLE
Cross-virtual dom signals

### ISSUE THREAD

**Issue by ealmloff** [DioxusLabs/dioxus#1569]
## Specific Demand

Signals currently use the context API to communicate with eachother. This works great within one virtual dom, but once you create a new virtual dom it can cause issues. For example, you cannot create a signal in one window and send it to another window in this [example](https://github.com/DioxusLabs/dioxus/blob/master/examples/multiwindow.rs)

## Implement Suggestion

We could switch from the context API to a global lazy static/oncecell runtime

**Comment by esimkowitz** [comment]
I've been thinking about this too as a way to even further signal-ify #1443

I would be interested in helping work on this!

**Comment by ealmloff** [comment]
I don't think this will be too difficult to implement. We need to change from the context API [here](https://github.com/DioxusLabs/dioxus/blob/master/packages/signals/src/rt.rs#L14) to a [unsync::OnceCell](https://docs.rs/once_cell/latest/once_cell/unsync/struct.OnceCell.html) in a [thread local](https://doc.rust-lang.org/std/macro.thread_local.html)

**Comment by esimkowitz** [comment]
Would thread local be enough for multi window support? I can see how that would help with multi-vdom, but it seems to me like a separate shared thread would be needed to manage signals between windows

**Comment by ealmloff** [comment]
> Would thread local be enough for multi window support? I can see how that would help with multi-vdom, but it seems to me like a separate shared thread would be needed to manage signals between windows
> 
> Update: oh are you thinking more in how to separate the ownership from the context, not for how to communicate signals between windows?

I think it would be enough for multi-window support, currently I believe we have desktop set up so all windows live on the same thread.

**Comment by rydb** [comment]
> Is there currently any workaround for using signals across multiple windows?

you could keep mutations to your data behind functions on newtype holders and have the value inside a rwlock

```rust 
struct DioxusValue<T> {
    value: Arc<RwLock<T>>,
}
```

**Comment by tdomhan** [comment]
fwiw, I'm successfully using signals across multiple windows created via [leak_with_caller](https://docs.rs/dioxus/latest/dioxus/prelude/struct.Signal.html#method.leak_with_caller), which works in my case as all signals live for the entirety of my application.
